src/Xamarin.Android.Tools.Bytecode/Kotlin/KotlinFixups.cs

```diff
@@ -253,17 +284,40 @@ static bool TypesMatch (TypeInfo javaType, KotlinType kotlinType, KotlinClass ko
 			// Sometimes Kotlin keeps its native types rather than converting them to Java native types
 			// ie: "Lkotlin/UShort;" instead of "S"
 			if (javaType.BinaryName.StartsWith ("L", StringComparison.Ordinal) && javaType.BinaryName.EndsWith (";", StringComparison.Ordinal)) {
-				if (KotlinUtilities.ConvertKotlinClassToJava (javaType.BinaryName.Substring (1, javaType.BinaryName.Length - 2)) == KotlinUtilities.ConvertKotlinTypeSignature (kotlinType, kotlinClass))
+				if (KotlinUtilities.ConvertKotlinClassToJava (javaType.BinaryName.Substring (1, javaType.BinaryName.Length - 2)) == KotlinUtilities.ConvertKotlinTypeSignature (kotlinType, kotlinFile))
 					return true;
 			}
 
 			// Same for some arrays
 			if (javaType.BinaryName.StartsWith ("[L", StringComparison.Ordinal) && javaType.BinaryName.EndsWith (";", StringComparison.Ordinal)) {
-				if ("[" + KotlinUtilities.ConvertKotlinClassToJava (javaType.BinaryName.Substring (2, javaType.BinaryName.Length - 3)) == KotlinUtilities.ConvertKotlinTypeSignature (kotlinType, kotlinClass))
+				if ("[" + KotlinUtilities.ConvertKotlinClassToJava (javaType.BinaryName.Substring (2, javaType.BinaryName.Length - 3)) == KotlinUtilities.ConvertKotlinTypeSignature (kotlinType, kotlinFile))
 					return true;
 			}
 
 			return false;
 		}
+
+		static string GetKotlinType (string jvmType, string kotlinClass)
+		{
+			// Handle erasure of Kotlin unsigned types
+			if (jvmType == "I" && kotlinClass == "kotlin/UInt;")
+				return "uint";
+			if (jvmType == "[I" && kotlinClass == "kotlin/UIntArray;")
```

Comment:
From the "obvious question is obvious" department -- transcending into the "oh god what horrors does this provoke?!" department -- is it possible to be given a `[[I?

There is no `kotlin.UIntArrayArray` type.  However, we "could" do:

```kotlin
	public open fun uintArray(value: Array<UIntArray>) {
	}
```

...which isn't actually special-cased by the compiler.  Meaning it "leaks", with `javap` showing!

```
  public void uintArray(kotlin.UIntArray[]);
```

WTF is a `kotlin.UIntArray` type?  (It's an actual type!)

So in *some* contexts, `UIntArray` becomes a JVM `int[]`, and in *other* contexts it remains a `UIntArray`.

I'm still not sure how to mentally process this.  I'm not sure what this should *mean* wrt binding at all.